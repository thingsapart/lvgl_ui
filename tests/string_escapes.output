DEBUG_PRINT: unmarshal_value: String 'Line1
Line2' falling back to IR_EXPR_LITERAL_STRING
Warning: Property 'tooltip_text' for object type 'label' (C var 'l_escapes') not found in API spec. Skipping.
DEBUG_PRINT: unmarshal_value: String 'Enter text here:
E.g., C:\Program Files' falling back to IR_EXPR_LITERAL_STRING
// Code generated by LVGL UI Generator

#include "lvgl.h"

// Forward declaration for any registered C pointers you might have from your application.
// Example: extern const lv_font_t my_custom_font;
void create_ui(lv_obj_t* parent) {
    {
        lv_obj_t* l_escapes = lv_label_create(parent);
        lv_label_set_text(l_escapes, "Line1\nLine2");
        registry_add_pointer(ui_registry, l_escapes, "label_escapes", "label");
    }
    {
        lv_obj_t* ta_escapes = lv_textarea_create(parent);
        lv_textarea_set_placeholder_text(ta_escapes, "Enter text here:\nE.g., C:\\Program Files");
        registry_add_pointer(ui_registry, ta_escapes, "ta_escapes", "textarea");
    }
}
