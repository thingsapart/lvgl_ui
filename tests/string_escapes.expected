Warning: Property 'tooltip_text' for object type 'label' (C var 'l_escapes') not found in API spec. Skipping.
// Code generated by LVGL UI Generator

#include "lvgl.h"

// Forward declaration for any registered C pointers you might have from your application.
// Example: extern const lv_font_t my_custom_font;
void create_ui(lv_obj_t* parent) {
    {
        lv_obj_t* l_escapes = lv_label_create(parent);
        lv_label_set_text(l_escapes, "Line1\nLine2");
        registry_add_pointer(ui_registry, l_escapes, "label_escapes", "label");
    }
    {
        lv_obj_t* ta_escapes = lv_textarea_create(parent);
        lv_textarea_set_placeholder_text(ta_escapes, "Enter text here:\nE.g., C:\\Program Files");
        registry_add_pointer(ui_registry, ta_escapes, "ta_escapes", "textarea");
    }
}
